kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/KeyCodec.java

```diff
@@ -0,0 +1,9 @@
+package dev.responsive.kafka.internal.db.mongo;
+
+import org.apache.kafka.common.utils.Bytes;
+
+interface KeyCodec {
```

Comment:
we should consider leveraging custom codecs in the Mongo Java driver (https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/data-formats/codecs/) so that we just automatically get this and don't forget using it somewhere.

I haven't explored whether this is possible yet - but if it's possible could be a worthwhile change

Reply:
I'll tackle this in a follow-up. It covers both kv and window stores.

Reply:
I'll tackle this in a follow-up. It covers both kv and window stores.